$MirOS: ports/graphics/png/patches/patch-scripts_makefile_openbsd,v 1.5 2008/04/07 19:37:25 tg Exp $
$OpenBSD: patch-scripts_makefile_openbsd,v 1.9 2004/07/06 21:53:16 brad Exp $
--- scripts/makefile.openbsd.orig	Thu May  8 11:57:47 2008
+++ scripts/makefile.openbsd	Tue May 13 16:27:52 2008
@@ -1,14 +1,18 @@
-# makefile for libpng
+# makefile for libpng under MirPorts
 # Copyright (C) 1995 Guy Eric Schalnat, Group 42, Inc.
 # Copyright (C) 2007 Glenn Randers-Pehrson
+# Copyright (C) 2008 Thorsten Glaser
 # For conditions of distribution and use, see copyright notice in png.h
 
 PREFIX?= /usr/local
-LIBDIR=	${PREFIX}/lib
+BINDIR=	${PREFIX}/bin
+LIBDIR=	${PREFIX}/pkgview/png/lib
 MANDIR= ${PREFIX}/man/cat
+INCDIR=	${PREFIX}/pkgview/png/include
+DOCDIR=	${PREFIX}/share/doc/png
 
-SHLIB_MAJOR=	0
-SHLIB_MINOR=	1.2.29
+SHLIB_MAJOR=	8
+SHLIB_MINOR=	1
 
 LIB=	png
 SRCS=	png.c pngerror.c pngget.c pngmem.c pngpread.c \
@@ -20,26 +24,48 @@ HDRS=	png.h pngconf.h
 CFLAGS+= -Wall
 CPPFLAGS+= -I${.CURDIR} -DPNG_NO_MMX_CODE 
 
+LDADD+=		-lz
+DPADD+=		${LIBZ}
+.if ${MACHINE_OS} != "Darwin"
+# Darwin does not need libm
+LDADD+=		-lm
+DPADD+=		${LIBM}
+.endif
+
 NOPROFILE= Yes
 
 CLEANFILES+= pngtest.o pngtest
+CLEANFILES+= libpng-config
 
 MAN=	libpng.3 libpngpf.3 png.5
-DOCS = ANNOUNCE CHANGES INSTALL KNOWNBUG LICENSE README TODO Y2KINFO libpng.txt
+DOCS = ANNOUNCE CHANGES INSTALL KNOWNBUG LICENSE README TODO Y2KINFO libpng-*.txt
 
+all: ${_LIBS} libpng-config
+
+libpng-config:
+	( cat scripts/libpng-config-head.in; \
+	  echo prefix=\"${PREFIX:Q}\"; \
+	  echo I_opts=\"-I${INCDIR:Q}\"; \
+	  echo L_opts=\"-L${LIBDIR:Q}\"; \
+	  echo R_opts=\"-Wl,-rpath,${LIBDIR:Q}\"; \
+	  echo libs=\"-lpng ${LDADD}\"; \
+	  cat scripts/libpng-config-body.in \
+	) >libpng-config
+	chmod +x libpng-config
+
 pngtest.o:	pngtest.c
 	${CC} ${CPPFLAGS} ${CFLAGS} -c ${.ALLSRC} -o ${.TARGET}
 
 pngtest:	pngtest.o
-	${CC} ${LDFLAGS} ${.ALLSRC} -o ${.TARGET} -L${.OBJDIR} -lpng -lz -lm
+	${CC} ${LDFLAGS} ${.ALLSRC} -o ${.TARGET} -L${.OBJDIR} -lpng ${LDADD}
 
-test:	pngtest
+test regress:	pngtest
 	cd ${.OBJDIR} && env \
-		LD_LIBRARY_PATH="${.OBJDIR}" ${.OBJDIR}/pngtest
+	    LD_LIBRARY_PATH=${.OBJDIR:Q} ${.OBJDIR}/pngtest
 
 beforeinstall:
-	if [ ! -d ${DESTDIR}${PREFIX}/include/libpng ]; then \
-	  ${INSTALL} -d -o root -g wheel ${DESTDIR}${PREFIX}/include/libpng; \
+	if [ ! -d ${DESTDIR}${INCDIR} ]; then \
+	  ${INSTALL} -d -o root -g wheel ${DESTDIR}${INCDIR}; \
 	fi
 	if [ ! -d ${DESTDIR}${LIBDIR} ]; then \
 	  ${INSTALL} -d -o root -g wheel ${DESTDIR}${LIBDIR}; \
@@ -53,21 +79,19 @@ beforeinstall:
 	if [ ! -d ${DESTDIR}${MANDIR}5 ]; then \
 	  ${INSTALL} -d -o root -g wheel ${DESTDIR}${MANDIR}5; \
 	fi
-	if [ ! -d ${DESTDIR}${PREFIX}/share/doc/png ]; then \
-	  ${INSTALL} -d -o root -g wheel ${DESTDIR}${PREFIX}/share/doc/png; \
+	if [ ! -d ${DESTDIR}${DOCDIR} ]; then \
+	  ${INSTALL} -d -o root -g wheel ${DESTDIR}${DOCDIR}; \
 	fi
 
 afterinstall:
 	@rm -f ${DESTDIR}${LIBDIR}/libpng_pic.a
 	@rm -f ${DESTDIR}${LIBDIR}/debug/libpng.a
-	@rm -f ${DESTDIR}${PREFIX}/include/png.h
-	@rm -f ${DESTDIR}${PREFIX}/include/pngconf.h
 	@rmdir ${DESTDIR}${LIBDIR}/debug 2>/dev/null || true
+	${INSTALL} ${INSTALL_COPY} -o ${BINOWN} -g ${BINGRP} \
+		-m ${BINMODE} libpng-config ${DESTDIR}${BINDIR}/
 	${INSTALL} ${INSTALL_COPY} -o ${SHAREOWN} -g ${SHAREGRP} \
-		-m ${NONBINMODE} ${HDRS} ${DESTDIR}${PREFIX}/include/libpng
+		-m ${NONBINMODE} ${HDRS} ${DESTDIR}${INCDIR}/
 	${INSTALL} ${INSTALL_COPY} -o ${SHAREOWN} -g ${SHAREGRP} \
-		-m ${NONBINMODE} ${HDRS} ${DESTDIR}${PREFIX}/include
-	${INSTALL} ${INSTALL_COPY} -o ${SHAREOWN} -g ${SHAREGRP} \
-		-m ${NONBINMODE} ${DOCS} ${DESTDIR}${PREFIX}/share/doc/png
+		-m ${NONBINMODE} ${DOCS} ${DESTDIR}${DOCDIR}/
 
 .include <bsd.lib.mk>
